### ISSUE TITLE
Which Kubernetes volume type should I use for segment cache?

### ISSUE THREAD

**Issue by henriquekops** [druid-io/druid-operator#233]
Hi team!

Do you have any advice on which kubernetes volume type is better / more appropriate for using as segment cache? (https://kubernetes.io/docs/concepts/storage/volumes/#volume-types)

I'm using GlusterFS for now, but I've seen in some issues that users are utilizing host/local paths...

Is there any pitfalls in my choice?

**Comment by AdheipSingh** [comment]
> utilizing host/local paths...
Yeah that has a lot of permutations and combinations to be taken care off. Since hostPath shall use a directory structure mounted on the node. Mutiple replica of the sts can get confusing.
In case of druid creating a pv  and pvc is the preferred since statefulsets are in use. You can try the local provisioner which uses local-ssds. druid is pretty perform-ant with local-ssds. Though as of now manual cleanup needs to be done in case a node goes down. ( in future operator can take care of this )

https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner

**Comment by henriquekops** [comment]
>  Though as of now manual cleanup needs to be done in case a node goes down.

@AdheipSingh you mean to cleanup the local volume to avoid data corruption?

Currently I'm statically provisioning volumes as well (since I don't want to delete segment cache when historical pod's PVCS goes down) but using an external service (GlusterFS) to control data redundancy and give flexibility for new volume creation.

My main concern is about latency issues, since I captured around 200ms (r/w) in my load tests when communicating with GlusterFS (in my infrastructure).

I think that local volumes should work well to bypass this problem, but I'm worried about this tighten relationship between k8s worker nodes and segment cache volumes, since I would need to download a large amount of data from S3 if a k8s node goes down.

**Comment by AdheipSingh** [comment]
> you mean to cleanup the local volume to avoid data corruption?

not exactly data corruption, i mean i haven't witnessed any. 

Mainly on cloud k8s, i have experienced, that pv cleanup is a manual process when a node is upgraded or goes down. Plus CA ( cluster autoscaler )does not work with local-ssd, so thats an issue too.

local-ssd are fast and performant. but running them isn't as smooth ( IMHO )
